manifests/compose.pp

```diff
@@ -7,118 +7,50 @@
 # @param version
 #   The version of Docker Compose to install.
 #
-# @param install_path
-#   The path where to install Docker Compose.
-#
-# @param symlink_name
-#   The name of the symlink created pointing to the actual docker-compose binary
-#   This allows use of own docker-compose wrapper scripts for the times it's
-#   necessary to set certain things before running the docker-compose binary
-#
-# @param proxy
-#   Proxy to use for downloading Docker Compose.
-#
-# @param base_url
-#   The base url for installation
-#   This allows use of a mirror that follows the same layout as the
-#   official repository
-#
-# @param raw_url
-#   Override the raw URL for installation
-#   The default is to build a URL from baseurl. If rawurl is set, the caller is
-#   responsible for ensuring the URL points to the correct version and
-#   architecture.
-#
-# @param curl_ensure
-#   Whether or not the curl package is ensured by this module.
-#
 class docker::compose (
-  Enum[present,absent] $ensure                 = present,
-  Optional[String]               $version      = $docker::params::compose_version,
-  Optional[String]               $install_path = $docker::params::compose_install_path,
-  Optional[String]               $symlink_name = $docker::params::compose_symlink_name,
-  Optional[Pattern['^((http[s]?)?:\/\/)?([^:^@]+:[^:^@]+@|)([\da-z\.-]+)\.([\da-z\.]{2,6})(:[\d])?([\/\w \.-]*)*\/?$']] $proxy = undef,
-  Optional[String]               $base_url     = $docker::params::compose_base_url,
-  Optional[String]               $raw_url      = undef,
-  Optional[Boolean]              $curl_ensure  = $docker::params::curl_ensure,
-) inherits docker::params {
-  if $facts['os']['family'] == 'windows' {
-    $file_extension = '.exe'
-    $file_owner     = 'Administrator'
-  } else {
-    $file_extension = ''
-    $file_owner     = 'root'
-  }
-
-  $docker_compose_location           = "${install_path}/${symlink_name}${file_extension}"
-  $docker_compose_location_versioned = "${install_path}/docker-compose-${version}${file_extension}"
-
-  if $ensure == 'present' {
-    if $raw_url != undef {
-      $docker_compose_url = $raw_url
-    } else {
-      $docker_compose_url = "${base_url}/${version}/docker-compose-${facts['kernel']}-${facts['os']['hardware']}${file_extension}"
+  Enum[present,absent] $ensure  = present,
+  Optional[String]     $version = undef,
+) {
+  if $docker::manage_package {
```

Comment:
We need to `include docker` before if we want to use `$docker::variable`.  That being said, what would be the point of including `docker::compose` if we don't want to install packages?  Can't we just get rid of the condition?

Reply:
The install.pp manifest looks pretty similar, essentially all wrapped in this `manage_package` condition. I've notice `Apt::Source['docker']` is only declared if `use_upstream_package_source` is true and `include apt` only happens if `manage_package` is true as well. `Yumrepo['docker']` is only declared if both `use_upstream_package_source` and `manage_package` are true.

Reply:
Yeh,  `docker::install` seems to have the same issue.  It should probably `include docker`, or maybe be a private class.

This module is not in a good shape unfortunately